File: frontend/features/chat/types/api/session-queue-item.ts
import type { InputFile } from "./session";

export interface SessionQueueItemResponse {
  queue_item_id: string;
  session_id: string;
  sequence_no: number;
  status: "queued" | "paused" | "promoted" | "canceled";
  prompt: string;
  permission_mode: string;
  attachments: InputFile[];
  client_request_id?: string | null;
  linked_run_id?: string | null;
  linked_user_message_id?: number | null;
  created_at: string;
  updated_at: string;
}

export interface SessionQueueItemUpdateRequest {
  prompt?: string | null;
  attachments?: InputFile[] | null;
}

File: backend/app/schemas/session_queue_item.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.input_file import InputFile


class SessionQueueItemResponse(BaseModel):
    """Persisted queued query response."""

    queue_item_id: UUID = Field(validation_alias="id")
    session_id: UUID
    sequence_no: int
    status: str
    prompt: str
    permission_mode: str
    attachments: list[InputFile] = Field(default_factory=list)
    client_request_id: str | None = None
    linked_run_id: UUID | None = None
    linked_user_message_id: int | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True, populate_by_name=True)


class SessionQueueItemUpdateRequest(BaseModel):
    """Queued query update request."""

    prompt: str | None = None
    attachments: list[InputFile] | None = None
